org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/IHasContext.java

```diff
@@ -0,0 +1,8 @@
+package org.eclipse.jdt.ls.debug.adapter;
+
+import java.util.Map;
+
+public interface IHasContext {
```

Comment:
Not quite sure about the naming. Is the context used by providers? If that's the case, then it's a common behavior across all providers. And a common IProvider is recommended.